### ISSUE TITLE
[LineChart]: need padding if legend is set to left or right position

### ISSUE THREAD

**Issue by scottdickerson** [carbon-design-system/carbon-charts#387]
#### I am submitting a...

- [ ] Feature request
- [ ] Design defect
- [x] Source code defect
- [ ] Demo/documentation defect
- [ ] Other

#### charts version:
0.16.24
### Issue description
Missing padding

#### Steps to produce the issue
Set the position of the Legend to the right and notice there's padding missing
#### Current behavior

Seems pretty clear that in legend.ts, padding is only added for Top and Bottom positions, not Left or Right
```
if (legendPosition === LegendPositions.BOTTOM || legendPosition === LegendPositions.TOP) {
			// TODO - Replace with layout component margins
			DOMUtils.appendOrSelect(svg, "rect.spacer")
				.attr("x", 0)
				.attr("y", 10)
				.attr("width", 20)
				.attr("height", 20)
				.attr("fill", "none");
		}
```

### Screenshot or recording
![image](https://user-images.githubusercontent.com/6663002/67621711-25c25480-f7d8-11e9-9014-954c4e624d6b.png)

**Comment by theiliad** [comment]
Coming in this week with the addition of Spacer components 🙂
